faust/transport/consumer.py

```diff
@@ -1097,6 +1097,18 @@ def _new_offset(self, tp: TP) -> Optional[int]:
                 acked.extend(gaps)
                 gap_for_tp[:gap_index] = []
             acked.sort()
+
+            # We iterate over it until we handle gap in the head of acked queue
+            # then return the previous committed offset.
+            # For example if acked[tp] is:
+            #    34 35 36 37
+            #  ^-- gap
+            # self._committed_offset[tp] is 31
+            # the return value will be: 31
+            if self._committed_offset[tp]:
+                if min(acked) - self._committed_offset[tp] > 0:
+                    return self._committed_offset[tp]
```

Comment:
We should return None here. Returning the currently committed offset would cause us to recommit the same value, but in that case there isn't really a need to commit anything.